subsys/bluetooth/audio/shell/bap_broadcast_assistant.c

```diff
@@ -498,8 +513,32 @@ static bool broadcast_source_found(struct bt_data *data, void *user_data)
 	char addr_str[BT_ADDR_LE_STR_LEN];
 	struct bt_uuid_16 adv_uuid;
 	uint32_t broadcast_id;
+	char broadcast_name[MAX_BROADCASTNAME_LEN + 1] = { 0 };
+	size_t broadcast_name_length = 0;
+	bool found_matched_broadcast_id = false;
+	bool found_matched_broadcast_name = false;
 	int err;
 
+	bt_addr_le_to_str(info->addr, addr_str, sizeof(addr_str));
+
+	/* Check for matched broadcast name (in case of add_by_broadcast_name) */
+	if (data->type == BT_DATA_BROADCAST_NAME) {
+		if (broadcast_name_scan.broadcast_name[0] != '\0') {
+			broadcast_name_length = (data->data_len < sizeof(broadcast_name)) ?
+				data->data_len : sizeof(broadcast_name) - 1;
+			memcpy(broadcast_name, data->data, broadcast_name_length);
+			if (!strncmp(broadcast_name, broadcast_name_scan.broadcast_name,
+					broadcast_name_length)) {
+				broadcast_name_scan.broadcast_name_is_matched = true;
```

Comment:
Is saving broadcast_name_is_matched and addr_str globally needed?

Reply:
In my implementation, they are used to find the broadcast_id of the same address (from the next advertisements).

Reply:
We can actually do the same as we do in the BAP broadcast assistant sample: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/bap_broadcast_assistant/src/main.c#L272-L377

i.e. we have stack-allocated struct the scan receive callback that is passed to the callback for each LTV. That way we do not need to store the information globally, while still being able to check for both values (and without potentially waiting for another advertising report as we do now)

Reply:
This solution also looks nice. I’ll try a quick and dirty implementation based on it and test it today. If I find it better and clearer, I’ll update my fix this way and push a new commit. Otherwise, I’ll keep the current implementation.